### PR TITLE
Enable Push Notifications

### PR DESCRIPTION
**Pull request by dishanest**
## Overview
Enable push notifications within Volume while preserving SwiftUI lifecycle. 

## Changes Made
- Add app capabilities: 
  - Push Notifications
  - Remote Notifications background mode
- Created `AppDelegate.swift`:
  - handles APNs registration callbacks
  - reference stored in `Main.swift` in `@UIApplicationDelegateAdaptor` wrapper so instance doesn't get deallocated
- `Main.swift` (on app startup):
  - request push notification permissions from user if not yet granted/denied
  - register for push notifications w/ APNs
  - made methods private to prevent app-level access from UI-level files

## Test Coverage
Followed [this tutorial by Apple](https://developer.apple.com/documentation/usernotifications/sending_push_notifications_using_command-line_tools) for sending test push notifications to an app via Terminal. DM me for the certificates & keys used for this test, or create your own on the [Developer Portal](https://developer.apple.com/account/resources/certificates/list). 

## Next Steps

Implement analytics (see below) for this feature. 

## Related PRs or Issues 

Re-implemented notifications support [changes by @camfruss](https://github.com/cuappdev/volume-ios/tree/cam/notifs) under SwiftUI paradigm instead of UIKit. Keeping his branch open for future reference on analytics for the notifications feature. 

## Screenshots 

IMG_5752

### CODE REVIEW

File: Volume/Supporting/Main.swift [hunk 1]
```diff
@@ -32,6 +34,19 @@ struct Main: App {
...
+    private func registerForPushNotifications() {
+        UNUserNotificationCenter.current().getNotificationSettings { settings in
+            if settings.authorizationStatus == .notDetermined {
+                UNUserNotificationCenter.current().requestAuthorization(options: [.alert, .badge, .sound]) { granted, error in
+                    print("Push notification permissions granted: \(granted)")
+                }
+            }
+        }
+        DispatchQueue.main.async {
+            UIApplication.shared.registerForRemoteNotifications()
```
Comment (danielVebman) [inline Volume/Supporting/Main.swift:47]: This should be inside of `getNotificationSettings`, both if notifs are authorized and after requesting authorization succeeds.

**Review by danielVebman — APPROVED** [review]
1 comment 🚀

File: Volume/Supporting/Main.swift [hunk 2]
```diff
@@ -32,6 +34,25 @@ struct Main: App {
             announcementPath: Secrets.announcementsPath
         )
     }
+    
+    private func registerForPushNotifications() {
+        UNUserNotificationCenter.current().getNotificationSettings { settings in
+            if settings.authorizationStatus == .notDetermined {
+                UNUserNotificationCenter.current().requestAuthorization(options: [.alert, .badge, .sound]) { granted, error in
+                    print("Push notification permissions granted: \(granted)")
```
Comment (spd75) [inline Volume/Supporting/Main.swift:42]: Not saying you should change this, but just wanted to verify you know this print statement is in here.  If you want to keep a print statement in production code that's up to you all, but pods I've been on previously try to limit print statements in production unless the feature will be worked on over a span of multiple PRs.

If you think you should keep it though go ahead.

File: Volume/Supporting/Main.swift [hunk 3]
```diff
@@ -12,9 +12,11 @@ import SwiftUI
...
 @main
 struct Main: App {
+    @UIApplicationDelegateAdaptor(AppDelegate.self) private var appDelegate
+    
     init() {
         configureFirebase()
-
+        registerForPushNotifications()
         let grayColor = UIColor(Color.volume.navigationBarGray)
         UINavigationBar.appearance().backgroundColor = grayColor
```
Comment (spd75) [inline Volume/Supporting/Main.swift:21]: Why do you define grayColor above?  Is it to prevent line 21 from getting too long?

**Review by spd75 — APPROVED** [review]
Looks great!  I just marked a few very minor things

**Pull request merged**